### DISCUSSION TITLE
How-to: Formulas to sum minutes to hours

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#2092] [Help] [answered]
I need to sum 15 min to the 10:38 (hh:mm), may you suggest which formulas I can use?
Normally I need to create the following attributes of a node: (I think to a 24-hour period for a day (and not to 2 periods of 12 hours (AM/PM))
Case 1)
name: from  --> and value: format(new Date(),'hh:mm') e.g.: 10:33
name: duration --> and value: 15
name: to --> and value: ??? this I do not know how to set, the expected result is 10:48
Case 2)
name: from  --> and value: format(new Date(),'hh:mm') e.g.: 10:33
name: duration --> and value: 75
name: to --> and value: ??? this I do not know how to set, the expected result is 11:48
Case 3)
name: from  --> and value: format(new Date(),'hh:mm') e.g.: 10:33
name: duration --> and value: 1515 (1day 1440 min + 75 min)
name: to --> and value: ??? this I do not know how to set, the expected result is +1d 11:48
Case 4)
name: from  --> and value: format(new Date(),'hh:mm') e.g.: 23:33
name: duration --> and value: 75
name: to --> and value: ??? this I do not know how to set, the expected result is +1d 00:48

Thanks for help

**Comment by Coffee-Search** [comment, marked as answer]
I think this is the formula you need - there could be more detailed error checking on the duration I force it to 0 with .num0 and discount it of it is 0 ... 

```
=
def from = node["from"].toString()
def duration = node["duration"].num0.toInteger()
def text = null


java.time.format.DateTimeFormatter timeFormatter = java.time.format.DateTimeFormatter.ofPattern("HH:mm")
java.time.LocalTime startTime

try {
    startTime = java.time.LocalTime.parse(from, timeFormatter)
} catch (Exception e) {
    text = "Invalid 'from' time format. Expected HH:mm."        
}
    
if (!text && duration < 1) {
    text = "Invalid 'duration'. Expected a positive integer."
}

if(!text) {
    java.time.LocalTime toTime = startTime.plusMinutes(duration)
    text=toTime.format(timeFormatter)

    int days = duration / (24 * 60)
    int remainingMinutes = duration % (24 * 60)
    java.time.LocalTime finalTime = startTime.plusMinutes(remainingMinutes)
    boolean crossesMidnight = finalTime.isBefore(startTime)
    if (crossesMidnight) {
        days += 1
    }
    
    def dayPart = days > 0 ? "+${days}d " : ""
    text = dayPart + finalTime.format(timeFormatter)

}

text
```

**Reply by Illegit** [reply]
Many thanks, it is exactly what I was in need of.
